### PR TITLE
Ssl tests

### PR DESCRIPTION
**Pull request by pepone**
This PR adds tests for using the macOS/iOS SSL APIs

### CODE REVIEW

File: cpp/src/Ice/SSL/SecureTransportUtil.h [hunk 1]
```diff
@@ -24,16 +25,16 @@ namespace Ice::SSL::SecureTransport
     //
     // Read certificate from a file.
     //
-    CFArrayRef loadCertificateChain(
+    ICE_API CFArrayRef loadCertificateChain(
```
Comment (pepone) [inline cpp/src/Ice/SSL/SecureTransportUtil.h:28]: I exported these methods for loading the certificates used by the tests.

File: cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp [hunk 2]
```diff
@@ -239,6 +239,14 @@ Ice::SSL::SecureTransport::TransceiverI::initialize(IceInternal::Buffer& readBuf
                     {
                         continue; // Call SSLHandshake to resume the handshake.
                     }
+                    else
+                    {
+                        throw SecurityException(
+                            __FILE__,
+                            __LINE__,
+                            "IceSSL: certificate verification failed. the certificate was explicitly rejected by the "
```
Comment (bernardnormier) [inline cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp:247]: I would remove the "explicitly". And capitalize after the period: The certificate was rejected...

**Review by bernardnormier — APPROVED** [review]
Looks good.

**Review by externl — APPROVED** [review]
Looks good to me!

**Pull request merged**